src/vs/workbench/parts/files/electron-browser/views/explorerViewer.ts

```diff
@@ -316,6 +320,27 @@ export class FileRenderer implements IRenderer {
 	}
 }
 
+function displayCurrentPath(inputBox: InputBox, initialRelPath: string, fileKind: FileKind) {
+	const value = inputBox.value;
+	if (inputBox.validate()) {
+		if (value && value.search(/[\\/]/) !== -1) {	// only show if there's a slash
+			const newPath = replaceWithNativeSep(paths.join(initialRelPath, value));
+			const fileType: string = FileKind[fileKind].toLowerCase();
+
+			inputBox.showMessage({
+				type: MessageType.INFO,
+				content: nls.localize('constructedPath', "Create **{0}** in **{1}**", fileType, newPath),
+				formatContent: true
+			});
+		}
+	}
+}
+
+function replaceWithNativeSep(str: string) {
```

Comment:
There should be a helper method in paths.ts so you do not need to introduce a new one. Try using what we alerady have.